lib/presentation/ui/home/home_page.dart

```diff
@@ -382,16 +430,53 @@ class _MapWidgetState extends State<MapWidget> {
           return IconButton(
             icon: Icon(
               Icons.place,
-              color: Colors.green,
+              color: selectedOccurrenceId != occurrence.id
+                  ? Colors.green
+                  : Colors.orange,
             ),
-            onPressed: () => print("clicked"),
+            onPressed: () {
+              selectedOccurrenceId = occurrence.id;
```

Comment:
Why are you using here both set state and putting data into the bloc? could you explain the logic?